src/sources/internal_logs.rs

```diff
@@ -26,41 +32,95 @@ pub struct InternalLogsConfig {
     /// By default, the [global `log_schema.host_key` option][global_host_key] is used.
     ///
     /// [global_host_key]: https://vector.dev/docs/reference/configuration/global-options/#log_schema.host_key
-    pub host_key: Option<String>,
+    #[serde(default)]
+    pub host_key: Option<OptionalValuePath>,
 
     /// Overrides the name of the log field used to add the current process ID to each event.
     ///
     /// The value will be the current process ID for Vector itself.
     ///
     /// By default, `"pid"` is used.
-    pub pid_key: Option<String>,
+    #[serde(default)]
+    pub pid_key: Option<OptionalValuePath>,
+
+    /// The namespace to use for logs. This overrides the global setting.
+    #[configurable(metadata(docs::hidden))]
+    #[serde(default)]
+    pub log_namespace: Option<bool>,
 }
 
 impl_generate_config_from_default!(InternalLogsConfig);
 
+impl InternalLogsConfig {
+    /// Generates the `schema::Definition` for this component.
+    fn schema_definition(&self, log_namespace: LogNamespace) -> Definition {
+        // `host_key` defaults to the `log_schema().host_key()` if it's not configured in the source.
+        let host_key = self
+            .host_key
+            .clone()
+            .map_or_else(
+                || parse_value_path(log_schema().host_key()).ok(),
+                |k| k.path,
+            )
+            .map(LegacyKey::Overwrite);
+
+        let pid_key = self
+            .pid_key
+            .clone()
+            .and_then(|k| k.path)
+            .map(LegacyKey::Overwrite);
+
+        // There is a global and per-source `log_namespace` config.
+        // The source config overrides the global setting and is merged here.
+        BytesDeserializerConfig
+            .schema_definition(log_namespace)
+            .with_standard_vector_source_metadata()
+            .with_source_metadata(
+                InternalLogsConfig::NAME,
+                host_key,
+                &owned_value_path!("host"),
+                Kind::bytes().or_undefined(),
+                Some("host"),
+            )
+            .with_source_metadata(
+                InternalLogsConfig::NAME,
+                pid_key,
+                &owned_value_path!("pid"),
+                Kind::integer(),
+                Some("pid"),
```

Comment:
```suggestion
                None,
```

I'd just do this as `None` since we haven't used/set the meaning elsewhere yet.